core/src/oauth/connection.rs

```diff
@@ -17,9 +17,45 @@ use serde::{Deserialize, Serialize};
 use std::str::FromStr;
 use std::time::Duration;
 use std::{env, fmt};
+use tracing::error;
 
 use super::providers::confluence::ConfluenceConnectionProvider;
 
+// Define the ErrorKind enum with serde attributes for serialization
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub enum ConnectionErrorCode {
```

Comment:
`thiserror`  (that we use [here](https://github.com/dust-tt/dust/blob/0e46f244ddf7d43669505440b3a2d124c7e686e8/core/src/sqlite_workers/client.rs#L23)) is quite nice to work with

Reply:
Will move forward as is but no need to pass arguments that much so not sure we need it yet?

Reply:
even without that, you can simply define variants and have the display automatically implemented (so no need to do one enum then one struct then one display impl)